### PR TITLE
feat(Masthead): Create base component

### LINKED ISSUE #136: Upgrade to Next.js 11.x.x

### LINKED ISSUE #153: Masthead (Presentational)
https://www.sketch.com/s/97e144f3-9c62-4ffd-8eca-94451b5536e1/a/25GdJk5

Screenshot 2021-07-13 at 10 52 27
Screenshot 2021-07-13 at 10 54 53

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #153
Closes #136

## Overview

Create `Masthead` component and add SVG support.

## Reason

>Create all presentational components in 'legacy' docs-site mocks.

## Work carried out

- [x] Upgrade to NextJS 11
- [x] Enable inline SVG support
- [x] Create base component
- [x] Add automated tests and stories

## Screenshot

Screenshot 2021-07-16 at 12 29 53
Screenshot 2021-07-15 at 12 11 29

### CODE REVIEW

**Comment by m7kvqbe1** [comment]
Waiting on this to implement `useDocumentClick` behaviour (clicking outside menu to close): https://github.com/Royal-Navy/design-system/pull/2592

File: components/presenters/Docs/Masthead/Masthead.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    <StyledMasthead>
+      <div>
+        <RNDSLogo />
+        <Badge variant="light">{version}</Badge>
+      </div>
+      <div>
+        {children}
+        <StyledMenuButton
+          type="button"
+          onClick={(e: React.MouseEvent<HTMLButtonElement>) => onToggle(e)}
```
Comment (thyhjwb6) [inline components/presenters/Docs/Masthead/Masthead.tsx:30]: This is the equivalent of:

```
onClick={onToggle}
```

File: components/presenters/Docs/Masthead/Masthead.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}) => {
+  return (
+    <StyledMasthead>
+      <div>
+        <RNDSLogo />
+        <Badge variant="light">{version}</Badge>
+      </div>
+      <div>
+        {children}
+        <StyledMenuButton
```
Comment (thyhjwb6) [inline components/presenters/Docs/Masthead/Masthead.tsx:28]: Is this required? Hence is `onToggle` required?

**Review by thyhjwb6 — COMMENTED** [review]
Looks good 👍 

Wondering if we need the toggle?

File: components/presenters/Docs/Masthead/Masthead.tsx (hunk 2 above)
Comment (m7kvqbe1) [reply components/presenters/Docs/Masthead/Masthead.tsx:28]: The toggle is there for the mobile navigation which is currently being implemented on another branch so it's kind of stubbed a bit.

**Review by thyhjwb6 — APPROVED** [review]
👍

**Pull request merged**